tests/Binance/SignerTests.cpp

```diff
@@ -4,202 +4,385 @@
-        "f0625dee"
-        "0a52"
-            "2a2c87fa"
-            "0a25""0a141d0e3086e8e4e0a53c38a90d55bd58b34d57d2fa120d0a03424e42108080e983b1de16"
-            "1225""0a146b571fc0a9961a7ddf45e49a88a4d83941fcabbe120d0a03424e42108080e983b1de16"
-        "126c"
-            "0a26"
-            "eb5ae987"
-            "21027e69d96640300433654e016d218a8d7ffed751023d8efe81e55dedbd6754c971"
-            "1240""8b23eecfa8237a27676725173e58154e6c204bb291b31c3b7b507c8f04e2773909ba70e01b54f4bd0bc76669f5712a5a66b9508acdf3aa5e4fde75fbe57622a1"
-            "2001"
-    );
-}
+    TEST(BinanceSigner, Sign) {
+        auto input = Proto::SigningInput();
+        input.set_chain_id("chain-bnb");
+        input.set_account_number(12);
+        input.set_sequence(35);
+        input.set_memo("");
+        input.set_source(1);
+
+        auto key = parse_hex("90335b9d2153ad1a9799a3ccc070bd64b4164e9642ee1dd48053c33f9a3a05e9");
+        input.set_private_key(key.data(), key.size());
+
+        auto &order = *input.mutable_trade_order();
+        auto result = Cosmos::Address::decode("bnb1hgm0p7khfk85zpz5v0j8wnej3a90w709vhkdfu");
+        ASSERT_TRUE(result.second);
+        auto keyhash = result.first.keyHash;
+        order.set_sender(keyhash.data(), keyhash.size());
+        order.set_id("BA36F0FAD74D8F41045463E4774F328F4AF779E5-36");
+        order.set_symbol("NNB-338_BNB");
+        order.set_ordertype(2);
+        order.set_side(1);
+        order.set_price(136350000);
+        order.set_quantity(100000000);
+        order.set_timeinforce(1);
+
+        auto signer = Binance::Signer(std::move(input));
+        auto signature = signer.sign();
+
+        ASSERT_EQ(hex(signature.begin(), signature.end()),
+                  "9123cb6906bb20aeb753f4a121d4d88ff0e9750ba75b0c4e10d76caee1e7d2481290fa3b9887a6225d69"
+                  "97f5f939ef834ea61d596a314237c48e560da9e17b5a");
+    }
+
+    TEST(BinanceSigner, Build) {
+        auto input = Proto::SigningInput();
+        input.set_chain_id("chain-bnb");
+        input.set_account_number(1);
+        input.set_sequence(10);
+
+        auto key = parse_hex("90335b9d2153ad1a9799a3ccc070bd64b4164e9642ee1dd48053c33f9a3a05e9");
+        input.set_private_key(key.data(), key.size());
+
+        auto &order = *input.mutable_trade_order();
+        auto address =
+                Cosmos::Address(HRP_BINANCE, parse_hex("b6561dcc104130059a7c08f48c64610c1f6f9064"));
+        auto keyhash = address.keyHash;
+        order.set_sender(keyhash.data(), keyhash.size());
+        order.set_id("B6561DCC104130059A7C08F48C64610C1F6F9064-11");
+        order.set_symbol("BTC-5C4_BNB");
+        order.set_ordertype(2);
+        order.set_side(1);
+        order.set_price(100000000);
+        order.set_quantity(1200000000);
+        order.set_timeinforce(1);
+
+        auto signer = Binance::Signer(std::move(input));
+        auto result = signer.build();
+
+        ASSERT_EQ(
+                hex(result.begin(), result.end()),
```

Comment:
can you use expected value as 1 hex string?

Reply:
+1, easier to copy